xdsl/dialects/tensor.py

```diff
@@ -37,6 +40,43 @@
 from xdsl.printer import Printer
 from xdsl.traits import NoMemoryEffect
 from xdsl.utils.exceptions import VerifyException
+from xdsl.utils.hints import isa
+
+
+@dataclass(frozen=True)
+class ContiguousArrayOfIntArray(AttrConstraint):
+    """
+    Enforce an ArrayAttr of ArrayAttr[IntegerAttr] to contain contiguous integer values in each inner array.
+    For example: [[0, 1], [2, 3]] is valid, but [[0, 2], [3, 4]] is not.
+    Each inner array must contain IntegerAttr elements whose integer values are contiguous.
+    An empty inner array is considered contiguous.
+    """
+
+    def verify(
+        self, attr: Attribute, constraint_context: ConstraintContext | None = None
+    ) -> None:
+        if not isa(attr, ArrayAttr[ArrayAttr[IntegerAttr]]):
+            raise VerifyException(
+                f"Expected ArrayAttr but got {getattr(attr, 'name', type(attr))}"
+            )
+
+        # Flatten all integer values from all inner arrays
+        flat_values: list[int] = []
+        for inner in attr.data:
+            elements = inner.data
+            if not elements:
+                continue
+            flat_values.extend(e.value.data for e in elements)
+        # Check that the flattened list is contiguous
+        for l, r in zip(flat_numbers[1:], flat_numbers[:-1]):
+            if flat_values[i + 1] != flat_values[i] + 1:
```

Comment:
```suggestion
            if r - l != 1:
```

Reply:
My bad

Reply:
resolved it a bit differently:
https://github.com/xdslproject/xdsl/pull/4626/commits/8a4d665fdea4e128104ac331387530010abbd601

but should be the same functionality